### PR TITLE
PubRepository grouping some constant/final values to facilitate future changes

### PR DESCRIPTION
**Pull request by antoninobajeli**
PubRepository grouping some constant/final values to facilitate future changes

### CODE REVIEW

File: examples/pub/lib/pub_repository.dart [hunk 1]
```diff
@@ -7,15 +7,21 @@ part 'pub_repository.g.dart';
 
 class PubRepository {
   final dio = Dio();
+  final scheme="https";
+  final host='pub.dartlang.org';
+  final packagesPath='api/packages';
+  final searchPath='api/search';
+  final likesPath='api/account/likes';
```
Comment (rrousselGit) [inline examples/pub/lib/pub_repository.dart:14]: I wouldn't have those constants, as these are specific to the API used. Some APIs have parts of their path in common, but that's more of an implementation detail than something we really want to share

File: examples/pub/lib/pub_repository.dart [hunk 2]
```diff
@@ -7,15 +7,21 @@ part 'pub_repository.g.dart';
 
 class PubRepository {
   final dio = Dio();
+  final scheme="https";
+  final host='pub.dartlang.org';
```
Comment (rrousselGit) [inline examples/pub/lib/pub_repository.dart:11]: Could you make those static private consts?

Also, this needs formatting and should use single quotes.

Comment (antoninobajeli) [reply examples/pub/lib/pub_repository.dart:11]: ok

Comment (antoninobajeli) [reply examples/pub/lib/pub_repository.dart]: I did it

File: examples/pub/lib/pub_repository.dart [hunk 3]
```diff
@@ -5,16 +5,21 @@ import 'package:pubspec_parse/pubspec_parse.dart';
 part 'pub_repository.freezed.dart';
 part 'pub_repository.g.dart';
 
+
```
Comment (rrousselGit) [inline examples/pub/lib/pub_repository.dart:8]:
```suggestion
```

File: examples/pub/lib/pub_repository.dart [hunk 4]
```diff
@@ -5,16 +5,21 @@ import 'package:pubspec_parse/pubspec_parse.dart';
...
 class PubRepository {
   final dio = Dio();
 
+  static String _scheme="https";
+  static String _host='pub.dartlang.org';
+
```
Comment (rrousselGit) [inline examples/pub/lib/pub_repository.dart:14]:
```suggestion
  static String _scheme = 'https';
  static String _host = 'pub.dartlang.org';

```

File: examples/pub/lib/pub_repository.dart [hunk 5]
```diff
@@ -8,13 +8,17 @@ part 'pub_repository.g.dart';
 class PubRepository {
   final dio = Dio();
 
+  static String _scheme = 'https';
+  static String _host = 'pub.dartlang.org';
+
+
```
Comment (rrousselGit) [inline examples/pub/lib/pub_repository.dart:14]:
```suggestion
```

File: examples/pub/lib/pub_repository.dart [hunk 6]
```diff
@@ -213,7 +216,7 @@ class Package with _$Package {
 @freezed
 class LikedPackage with _$LikedPackage {
   factory LikedPackage({required String package, required bool liked}) =
-      _LikedPackage;
+  _LikedPackage;
```
Comment (rrousselGit) [inline examples/pub/lib/pub_repository.dart:219]:
```suggestion
      _LikedPackage;
```

File: examples/pub/lib/pub_repository.dart [hunk 7]
```diff
@@ -222,7 +225,7 @@ class LikedPackage with _$LikedPackage {
 @freezed
 class LikedPackagesResponse with _$LikedPackagesResponse {
   factory LikedPackagesResponse({required List<LikedPackage> likedPackages}) =
-      _LikesPackagesResponse;
+  _LikesPackagesResponse;
```
Comment (rrousselGit) [inline examples/pub/lib/pub_repository.dart:228]:
```suggestion
      _LikesPackagesResponse;
```

File: examples/pub/lib/pub_repository.dart [hunk 8]
```diff
@@ -8,13 +8,16 @@ part 'pub_repository.g.dart';
 class PubRepository {
   final dio = Dio();
 
+  static String _scheme = 'https';
+  static String _host = 'pub.dartlang.org';
```
Comment (rrousselGit) [inline examples/pub/lib/pub_repository.dart:12]:
```suggestion
  static String _scheme = 'https';
  static String _host = 'pub.dartlang.org';

  final dio = Dio();
```

**Review by rrousselGit — APPROVED** [review]

**Pull request merged**